### ISSUE TITLE
Disabling UMR for module

### ISSUE THREAD

**Issue by battaglia01** [spyder-ide/spyder#17358]
This is either a feature request, or a how-to, depending on if there's some way to do this.

Suppose I've written a custom module called `mymodule`, and for whatever reason it doesn't play nice with UMR. I would like to disable UMR for `mymodule`.

Then I can go into the preferences and click "Set UMR excluded (not reloaded) modules" and try to add `mymodule`, but I will get the following popup:

<img width="532" alt="image" src="https://user-images.githubusercontent.com/1020594/154479061-c6191a88-8e4e-4604-bb61-215be912b57f.png">

I would guess that this is because the UMR is looking in site-packages, or something in PATH, etc for this module, whereas this is just something I've written in the current working directory.

''Is there some way to disable UMR for some module I've written, which is just in the pwd and not installed as a site particular module?''

I'm not quite sure how related this is to #2451. It seems that the __umr__ variable was never added, but the "UMR excluded modules" feature in the preferences would basically be the way to address #2451 (e.g. you could put in "numba" as the person requested). However, I'm not sure if there's some other way to do this.

**Comment by ccordoba12** [comment]
Hey @battaglia01, thanks for reporting. So, you're asking for excluding user modules too? I think that defeats the purpose of the UMR (which stands for User Module Reloader), doesn't it?

**Comment by mbinfinitone** [comment]
Correct. In particular, in this situation, I have a config module with shared global variables that several modules are using, so that they share state between the modules within those few variables. Then, for whatever reason, the UMR is causing that state to reset, so the different modules importing it have different ideas of what the shared state is. I guess that could be its own bug report although I am not quite sure what the cause is.

As a simple workaround I've now just disabled UMR, but it would be nice to be able to leave it enabled for other user modules besides this one.

**Comment by ccordoba12** [comment]
I think this is a good proposal, and something we haven't considered before. We'll try to do it in a future release.

**Comment by mrclary** [comment]
I've seen this issue before as well. I think the problem is that Spyder does not find the module in the current search paths (e.g. PYTHONPATH Manager). I've seen this before when a module is in a project directory, I add it to the UMR exclude list, and all is okay, but if I change projects then there is a problem because the project path is no longer in PYTHONPATH Manager.

So I think all we should need to do is let Spyder keep the list of excluded modules in the configuration and conditionally exclude them at module reload time.

**Comment by mrclary** [comment]
So there seems to be some idiosyncrasy here as well.
If I add the path to a local module to PYTHONPATH Manager, I can add the module to the UMR exclusion list without issue. If I remove the path from PYTHONPATH Manager, I can still remove/add the module to the UMR exclusion list without issue. However, upon restarting Spyder, I can no longer add it to the UMR exclusion list. Nevertheless, I think the solution remains the same.

**Comment by battaglia01** [comment]
OK, that makes sense. So the workaround, for now, is to let the UMR think that it isn't a "user module" by adding the path to the PYTHONPATH. So far it looks as though it works even for complex packages, e.g. I can add `util.shared.blah` to the UMR exclusion list as long as the "util" folder is within something in the PYTHONPATH. Is this supposed to also work if I do `from util.shared.blah import blah`?

**Comment by mrclary** [comment]
> Is this supposed to also work if I do `from util.shared.blah import blah`?

I believe so.
